analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+        \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+      ) |> 
+  # add in marker gene expression to dataframe 
+  dplyr::left_join(marker_gene_exp, by = "barcodes") |> 
+  # combine all genes into a single column for easy faceting
+  tidyr::pivot_longer(
+    cols = starts_with("ENSG"),
+    names_to = "ensembl_gene_id",
+    values_to = "gene_expression"
+  ) |> 
+  # join with marker gene df to get gene symbols for plotting 
+  dplyr::left_join(marker_genes_df, by = c("ensembl_gene_id")) |>
+  # get rid of excess columns
+  dplyr::select(barcodes, UMAP1, UMAP2, gene_symbol, ensembl_gene_id, gene_expression) |> 
+  # add in marker gene classifications
+  dplyr::left_join(marker_gene_results_df) |> 
+  # add in copykat results 
+  dplyr::left_join(copykat_results_df) |> 
+  # add in infercnv results and metadata
+  dplyr::left_join(infercnv_results_df) |> 
+  dplyr::left_join(infercnv_metadata_df) |> 
+  # gene set scores 
+  dplyr::left_join(geneset_scores_df)
+
+```
+
+## Identify tumor cell population 
+
+Below is a UMAP showing which cells are labeled as tumor cells using each of the various classification methods. 
+
+```{r}
+# plot tumor cells for each classification method 
+tumor_umap_df <- coldata_df |> 
+  dplyr::select(barcodes, UMAP1, UMAP2, marker_gene_classification, copykat_classification, cnv_proportion_classification) |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  )
+
+ggplot(tumor_umap_df, aes(x = UMAP1, UMAP2, color = classification)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  facet_wrap(vars(method))
+```
+
+This plot shows us that there is a pretty clear consensus that most of the cells in the bottom left group are tumor cells. 
+From the original marker gene analysis we know that not all marker genes are expressed and that there is quite a bit of variance in expression. 
+Because of that and the fact that both CNV inference methods show similar annotations, we will classify any cells that are labeled as tumor by both `CopyKAT` and `InferCNV` as tumor cells. 
+
+```{r}
+# create a combined classification between inferCNV and copyKAT
+# cells that are classified as tumor by both are kept as tumor, all other are normal 
+coldata_df <- coldata_df |> 
+  dplyr::mutate(
+    combined_cnv_classification = dplyr::if_else(
+      cnv_proportion_classification == "Tumor" & copykat_classification == "Tumor",
+      "Tumor",
+      "Normal"
+    ),
+    # there are cells classified as NA by copykat, we want those to be normal 
+    combined_cnv_classification = dplyr::if_else(is.na(copykat_classification) & is.na(combined_cnv_classification), "Normal", combined_cnv_classification)
+  )
+```
+
+```{r}
+# look at which cells are tumor/normal
+# be sure to remove any duplicates 
+coldata_df |> 
+  dplyr::select(UMAP1, UMAP2, barcodes, combined_cnv_classification) |>
+  unique() |> 
```

Comment:
Faster, apparently. But a bigger question is why are there duplicates? (got it..., see other comments)
```suggestion
  dplyr::distinct() |> 
```